exchanges/gateio/gateio_wrapper.go

```diff
@@ -2579,3 +2560,224 @@ func (g *Gateio) GetCurrencyTradeURL(_ context.Context, a asset.Item, cp currenc
 		return "", fmt.Errorf("%w %v", asset.ErrNotSupported, a)
 	}
 }
+
+// WebsocketSubmitOrder submits an order to the exchange
+// NOTE: Regarding spot orders, fee is applied to purchased currency.
+func (g *Gateio) WebsocketSubmitOrder(ctx context.Context, s *order.Submit) (*order.SubmitResponse, error) {
+	err := s.Validate(g.GetTradingRequirements())
+	if err != nil {
+		return nil, err
+	}
+
+	s.Pair, err = g.FormatExchangeCurrency(s.Pair, s.AssetType)
+	if err != nil {
+		return nil, err
+	}
+	s.Pair = s.Pair.Upper()
+
+	switch s.AssetType {
+	case asset.Spot:
+		req, err := g.getSpotOrderRequest(s)
+		if err != nil {
+			return nil, err
+		}
+
+		got, err := g.WebsocketSpotSubmitOrder(ctx, req)
+		if err != nil {
+			return nil, err
+		}
+		return g.deriveSpotWebsocketOrderResponse(got)
+	case asset.Futures:
+		amountWithDirection, err := getFutureOrderSize(s)
+		if err != nil {
+			return nil, err
+		}
+
+		timeInForce, err := getTimeInForce(s)
+		if err != nil {
+			return nil, err
+		}
+
+		a, err := getAssetFromFuturesPair(s.Pair)
```

Comment:
I think we missed this; We have s.Asset, so we should use that.
We're going to validate it anyway inside `WebsocketFuturesSubmitOrders` against the contract.